solr/core/src/java/org/apache/solr/cloud/api/collections/CreateCollectionCmd.java

```diff
@@ -321,6 +319,8 @@ public void call(ClusterState clusterState, ZkNodeProps message, @SuppressWarnin
         }
       }
       if(isPRS) {
+        byte[] data = Utils.toJSON(Collections.singletonMap(collectionName, clusterState.getCollection(collectionName)));
+        ocmh.zkStateReader.getZkClient().setData(collectionPath, data, true);
         ocmh.overseer.submit(new RefreshCollectionMessage(collectionName));
       }
 
```

Comment:
yes. How else do we build that {{Map}} in case of PRS ?

Reply:
it seems we don't need special case here as else part is doing for non-prs?

Reply:
> > @noblepaul @chatman would be good avoid special `prs` cases, if not require. Let me what you think about it?
> 
> Yes, I would like to avoid as many special cases as possible. But PRS requires special handling



> > @noblepaul @chatman would be good avoid special `prs` cases, if not require. Let me what you think about it?
> 
> Yes, I would like to avoid as many special cases as possible. But PRS requires special handling



> > @noblepaul @chatman would be good avoid special `prs` cases, if not require. Let me what you think about it?
> 
> Yes, I would like to avoid as many special cases as possible. But PRS requires special handling

Yeah. Can we look the cases which I mentioned above!